pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/StartControllerCommand.java

```diff
@@ -161,6 +161,10 @@ public boolean execute()
     Map<String, Object> properties = new HashMap<>();
     if (_configFileName != null) {
       properties.putAll(PinotConfigUtils.generateControllerConf(_configFileName));
+      // Override the zkAddress and clusterName to ensure ServiceManager is connecting to the right Zookeeper and Cluster.
+      // Configs existence is already verified.
+      _zkAddress = properties.get(ControllerConf.ZK_STR).toString();
+      _clusterName = properties.get(ControllerConf.HELIX_CLUSTER_NAME).toString();
```

Comment:
couldn't you simply remove the `setZkAddress()` etc calls above, since the properties get built here either way?

Reply:
same for the others

Reply:
Not sure if this will break others who wrapper this command class.
Also, we only update these two configs when users start pinot using the config file.